youtube_dl/extractor/iol.py

```diff
@@ -0,0 +1,98 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+from .common import InfoExtractor
+
+import re
+
+
+class IOLIE(InfoExtractor):
+    _VALID_URL = r'http://(tviplayer|(www\.tvi24))\.iol\.pt/.*/(?P<id>[0-9a-f]{24})($|\/)'
```

Comment:
maisfutebol.iol.pt is simular to tvplayer.iol.pt and tvi24.iol.pt.
to detect only urls with videos,may be it's better to check that `/video/` or `/videos/` in the url.